### DISCUSSION TITLE
How to install on Raspberry Pi Pico W?

### DISCUSSION THREAD

**Discussion by lordofscripts** [tinygo-org/tinygo#4067] [Q&A] [answered]
I am thinking of buying a π Pico W and, even though I learned Python, I would prefer to program it with Go. What would be the way to flash it on the π Pico W?

And, does it replace the MicroPython interpreter or is it installed side-by-side?

**Comment by marekor555** [comment, marked as answer]
it replaces micropython interpreter, but you can always reflash micropython onto it. Though there are some problems with builtin led of pi Pico W, because it's controlled by the CYW chip (the wifi one). For wifi check: https://github.com/soypat/cyw43439. But, remember the support for wifi chip isn't official (yet).

**Reply by marekor555** [reply]
PS: I have pi pico W. It's a great controller.

**Comment by andrewfstratton** [comment]
You can use VSCode with a tiny go extension and the experience is very like coding Go - though I haven't tried debugging myself - I have used serial print for debugging.

You can either just build and auto deploy, or build a UF2 which you then drop into a pico after powering on with bootsel held down.

I have built for several RP2040 boards now without issues, including Rpi Pico W, Badger2040, Waveshare RP2040 zero and Maker Pi 2040.

For Pico I would recommend a Switchable USB hub since it doesn't have a reset button - many other boards have reset or on/off switch.

BTW - as already noted - the Pico W Led doesn't work the same as the standard Pico - which is a pain...so if you don't need wifi/bluetooth it might be simpler to stick with the Pico.
